eth/vm/computation.py

```diff
@@ -319,7 +323,41 @@ def stack_pop(self, num_items: int=1, type_hint: str=None) -> Any:
         Raise `eth.exceptions.InsufficientStack` if there are not enough items on
         the stack.
         """
-        return self._stack.pop(num_items, type_hint)
+        if num_items == 1:
```

Comment:
I'd like to completely get rid of the type hinting and any conversion between `int <-> bytes` in both the `Computation` and `Stack` APIs, moving **all** conversion down into the opcode functions themselves.

For example:

https://github.com/ethereum/py-evm/blob/dfe43ae8031767837eaf04316c0052f7af02db4c/eth/vm/logic/memory.py#L6-L8

This would become:

```python
def mstore(computation: BaseComputation) -> None:
    start_position = computation.stack_pop(type_hint=constants.UINT256)
    raw_value = computation.stack_pop1()
    value = int_to_big_endian(raw_value)
    ...
```